bash/create-service-principal-2.0.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
```

Comment:
I think we would see more value if we added this to the original script and automatically detected whether the 1.0 or 2.0 cli is installed. For example, `if (command -v az >/dev/null)` returns true if the 2.0 cli is installed. That also lets us leverage the existing subscription selection logic rather than copying it over to this new script